src/expr/mod.rs

```diff
@@ -134,98 +138,118 @@ impl Function {
         }
     }
 
-    /// Returns the `DataType` of a column filtered by the current `Function`
+    /// Returns the `Schema` filtered by the current `Function`
     ///
     /// # Arguments:
-    /// * `column` - The `Column` to be filtered
-    /// * `datatype` - The `DataType` of `column`
-    ///
-    /// For example, we consider the `Column` `my_col` with `datatype = float[0 10]`.
-    ///     -  `self = gt(my_col, 5)` returns `DataType` `float[5 10]`,
-    ///     -  `self = gt(my_col, -5)` returns `DataType` `float[0 10]`,
-    ///     -  `self = gt(another_col, 5)` returns `DataType` `float[0 10]`
+    /// * `schema` - The `Schema` to be filtered
     ///
     /// Note: for the moment, we support only `Function` made of the composition of:
     /// - `Gt`, `GtEq`, `Lt`, `LtEq` functions comparing a column to a float or an integer value,
     /// - `Eq` function comparing a column to any value,
     /// - `And` function between two supported Expr::Function,
     /// - 'InList` test if a column value belongs to a list
-    // TODO: OR
-    pub fn filter_column_data_type(&self, column: &Column, datatype: &DataType) -> DataType {
+    // TODO : OR
+    pub fn filter_schema(&self, schema: &Schema) -> Result<Schema> {
```

Comment:
The code is structured such that:

- `data_type` is low level and independent
- `expr` is higher level and depends on `data_type` (it enables more or less composition of objects in `DataType`)
- `relation` is the main object manipulated in qrlew and depends on the lower level `expr` and `data_type`

Can you make sure this structure is preserved? (fix when it is not)

This means this code should probably live in the `relation::schema` module or `relation`.